### ISSUE TITLE
FetchError: Unexpected token < in JSON at position 0

### ISSUE THREAD

**Issue by noskovvkirill** [supabase/supabase#4464]
# Bug report

## Describe the bug

Hi guys! I'm using Supabase in one of my projects and I face a really weird issue. First I thought it's related to the JS version of the superbase-js library (some people reported the same error). Updating the client fixed that, but then the issue reappeared again.
I see
`error {
  message: 'FetchError: invalid json response body at https://tcmqmkigakxeiuratohw.supabase.co/rest/v1/mirroritems_test?select=*&order=timestamp.desc.nullslast&limit=19 reason: Unexpected token < in JSON at position 0',
  details: '',
  hint: '',
  code: ''
}`

## To Reproduce
I use the alpha preview of functions. The last time Supabase was down it coincide with the function that was triggered. Not sure if it caused an issue or not. If I mock the data and trigger the function myself, it works correctly. 
I don't have a clue how to reproduce it exactly, it seems to me that it's not the issue with the client or framework I use. But just in case,  I use next.js "11.1.2" with  "@supabase/supabase-js": "^1.28.5",

Restarting the server seems to fix the issue. 

## Expected behavior

Stable runtime. 

## Screenshots

<img width="763" alt="issue" src="https://user-images.githubusercontent.com/50060130/145830719-8b279a53-9bdb-4b1e-b485-a145496fe2c7.png">

## System information

- OS: macOS
- Version of supabase-js  1.28.5
- Version of Node.js: v14.15.1

**Comment by kiwicopple** [comment]
very odd - it seems that it's trying to decode the response but the response is not being returned as JSON. Are you able to give us some steps to reproduce this @noskovvkirill ?

**Comment by noskovvkirill** [comment]
@kiwicopple  Not really, not sure what is happening exactly. I will try to monitor it and see if I can reproduce it. It happened two times within the last couple of days. I thought it's related to https://github.com/supabase/supabase-js/pull/305, but the client update didn't fix that.

**Comment by noskovvkirill** [comment]
It happens again 😅 This time database reload doesn't help. I've tried to fetch different tables, but the issue seems to appear independent of the content! API page is not displayed on Supabase own front-end.
<img width="1417" alt="Снимок экрана 2021-12-15 в 13 46 59" src="https://user-images.githubusercontent.com/50060130/146189309-59b78704-a24e-4a05-9953-c7a522d2bfa0.png">

**Comment by kiwicopple** [comment]
@noskovvkirill - can you please submit a support ticket here: https://app.supabase.io/support/new

Thanks!

**Comment by 0hq** [comment]
Having a similar issue.

**Comment by edwardz8** [comment]
Having the same issue

**Comment by timuster** [comment]
Turning off RLS had no effect - still running into the issue randomly.

**Comment by dragarcia** [comment]
Hi all 👋 

In order to get a more helpful error response, we have recently updated our JS libraries (to get in the fixes for this [issue](https://github.com/supabase/postgrest-js/issues/241) in the `supabase/postgrest-js` library via this [PR](https://github.com/supabase/postgrest-js/pull/242)). Would you mind taking the following steps to upgrade it:
1. Run `npm run update` within your directory.
2. `@supabase/postgrest-js` should then now be in version 0.35.1 or above. To verify this, you can run `npm list @supabase/postgrest-js`.

This won't solve the issue at hand but at the very least we should now get a less vague, and hopefully more insightful response than `Unexpected token < in JSON at position 0`, and should eventually direct all of us to the root cause/s.

**Comment by kiwicopple** [comment]
I'll close this for now as this issue should be solved on the latest library version

**Comment by satyapalsharma** [comment]
Got the same issue yesterday around 11:20 IST

**Comment by redareda9** [comment]
Got the same issue today

**Comment by thuongdinh** [comment]
I got the same issue today

**Comment by steve-chavez** [comment]
FYI, we're tracking this one in https://github.com/supabase/supabase/issues/6839

**Comment by miketromba** [comment]
Just experienced the same issue and it caused my server to crash.
I believe it happened as a result of calling:
```js
const { user, error } = await supabase.auth.api.getUser(token)
```
```js
// Error message:
// FetchError: invalid json response body at https://fmapsufvoocumtvhyody.supabase.co/auth/v1/user 
// reason: Unexpected token < in JSON at position 0

// This line within node-fetch threw the error:
// return Body.Promise.reject(new FetchError(`invalid json response body at ${_this2.url} reason: ${err.message}`, 'invalid-json'));
```
While the root of this bug is being fixed, in the meantime is there some way to capture the error and reveal it in the returned .error prop of the supabase response object instead of allowing the supabase SDK method to throw? 

I assume most codebases will not expect supabase methods to throw... this is what caused my server to crash. If the error was passed along via the error prop, my code would have handled it properly.

**Comment by icopp** [comment]
I've repeatedly run into this problem while trying to do bulk operations on `auth.api.getUser` / `auth.api.createUser` to synchronize a new Supabase project with a (for now) external source of truth on users. Even rate-limiting my operations to no more than 20 in parallel, I have to put extensive retry logic on anything that touches `auth.api`.

I seriously don't understand what it's doing, because at this point I could probably just start direct inserting into the `users` table instead and have more success, and I've already started using a view on `users` as a workaround instead of using `getUser`.

**Comment by steve-chavez** [comment]
@miketromba @icopp Though the error message is the same, the issue you mention comes from the Auth API(GoTrue) and that's not used by `postgrest-js`.

Perhaps you could open an issue on https://github.com/supabase/gotrue-js/ for better visibility?

**Comment by J0** [comment]
This seems to be a class of errors rather than a single isolated error some of which arise from the createUser/getUser endpoints. Moving to Auth repo to triage
